src/app/component/ProductLinks/ProductLinks.container.js

```diff
@@ -28,6 +28,36 @@ export class ProductLinksContainer extends PureComponent {
         linkType: PropTypes.string.isRequired
     };
 
+    state = {
+        siblingsHaveBrands: false,
+        siblingsHavePriceBadge: false,
+        siblingsHaveTierPrice: false,
+        siblingsHaveConfigurableOptions: false
+    };
```

Comment:
Here we already have access to the products, don't we? Maybe it'd be better to calculate this state here instead of doing that down the render tree in order to omit extra rerenders? That could save us a few reconciliation cycle calls.

Reply:
This is a great idea, but it requires a lot if refactoring, since some values have complex calculation methods encapsulated inside the container classes.
It's easier to set appropriate state if the element is actually rendered in the product card.

Reply:
Okay then.